gamepage_resize.py

```diff
@@ -0,0 +1,490 @@
+            self.button("settings", 515, 20, 70, 20, tan_color, tan_highlight, 'settings')
+            self.image.blit(render_text(15, "Settings", black_color), (520, 18))
+
+    def load_intro_buttons(self):
+        self.button("player1", 125, 300, 150, 34, tan_color, tan_highlight, 'player1')
+        self.image.blit(render_text(28, "1-Player", black_color), (150, 295))
+        self.button("player2", 325, 300, 150, 34, tan_color, tan_highlight, 'player2')
+        self.image.blit(render_text(28, "2-Player", black_color), (350, 295))
+
+    def load_settings_buttons(self):
+        draw_red_circle(265, 385, settings_circle_radius, settings_outline_radius, self.image)
+        draw_black_circle(335, 385, settings_circle_radius, settings_outline_radius, self.image)
+        self.button("soundOn", 100, 150, 60, 20, tan_color, tan_highlight, 'soundOn')
+        self.image.blit(render_text(12, "Sound On", black_color), (103, 152))
+        self.button("soundOff", 175, 150, 60, 20, tan_color, tan_highlight, 'soundOff')
+        self.image.blit(render_text(12, "Sound Off", black_color), (177, 152))
+        self.button("listOnline", 350, 150, 150, 30, tan_color, tan_highlight, 'listOnline')
+        self.image.blit(render_text(20, "List me online", black_color), (365, 150))
+        self.button("2 player page", 350, 185, 150, 30, tan_color, tan_highlight, 'player2')
+        self.image.blit(render_text(20, "2 player page", black_color), (365, 185))
+        pg.draw.rect(self.image, tan_color, (210, 230, 180, 30))
+        self.image.blit(render_text(20, "Update Username", black_color), (215, 230))
+        pg.draw.rect(self.image, tan_color, (210, 270, 180, 30))
+        self.image.blit(render_text(20, "Update Password", black_color), (215, 270))
+
+    def load_1p_options_button(self):
+        draw_red_circle(265, 385, settings_circle_radius, settings_outline_radius, self.image)
+        draw_black_circle(335, 385, settings_circle_radius, settings_outline_radius, self.image)
+        self.button("easy", 225, 150, 150, 30, tan_color, tan_highlight, 'main1')
+        self.image.blit(render_text(24, "Easy", black_color), (270, 145))
+        self.button("medium", 225, 205, 150, 30, tan_color, tan_highlight, 'main1')
+        self.image.blit(render_text(24, "Medium", black_color), (260, 200))
+        self.button("hard", 225, 260, 150, 30, tan_color, tan_highlight, 'main1')
+        self.image.blit(render_text(24, "Hard", black_color), (270, 255))
+
+    def load_2p_options_button(self):
+        self.button("LocalGame", 100, 150, 150, 30, tan_color, tan_highlight, 'main2')
+        self.image.blit(render_text(28, "Local Game", black_color), (105, 145))
+        self.button("PlayOnline", 350, 150, 150, 30, tan_color, tan_highlight, 'PlayOnline')
+        self.image.blit(render_text(28, "Play Online", black_color), (355, 145))
+
+        xNum = 350
+        yNum = 200
+        if self.client and hasattr(self.client, "otherPlayers"):
```

Comment:
I think I know why the other players aren't showing up in this version -- it looks like you're rendering the static background visuals first, and then adding all the buttons second. The list of otherPlayers, and the incoming challenges, are dynamic -- they can come in after the page loads, and they'll need to cause updates to the 2 player page.

The original file was using while loops, so those dynamic changes were captured. Calling `self.client.pump()` might put a bandaid on the problem, but there needs to be some sort of loop that loads the page.